ports/nrf52840/core/hardware/bma400/bma400.cpp

```diff
@@ -0,0 +1,397 @@
+    m_bma400_sensor_conf.type = BMA400_ACCEL;
+    /* Modify the desired configurations as per macros
+     * available in bma400_defs.h file */
+    m_bma400_sensor_conf.param.accel.odr = BMA400_ODR_100HZ;
+    m_bma400_sensor_conf.param.accel.range = BMA400_RANGE_16G;
+    m_bma400_sensor_conf.param.accel.data_src = BMA400_DATA_SRC_ACCEL_FILT_2;
+
+    /* Set the desired configurations to the sensor */
+    rslt = bma400_set_sensor_conf(&m_bma400_sensor_conf, 1, &m_bma400_dev);
+    bma400_check_rslt("bma400_set_sensor_conf", rslt);
+
+    // note: Sleep mode: Registers readable and writable, no sensortime
+    rslt = bma400_set_power_mode(BMA400_MODE_SLEEP, &m_bma400_dev); 
+    bma400_check_rslt("bma400_set_power_mode", rslt);
+
+
+    m_bma400_device_conf.type = BMA400_INT_PIN_CONF;
+    m_bma400_device_conf.param.int_conf.int_chan = BMA400_INT_CHANNEL_1;
+    m_bma400_device_conf.param.int_conf.pin_conf = BMA400_INT_PUSH_PULL_ACTIVE_0;
+    rslt = bma400_set_device_conf(&m_bma400_device_conf, BMA400_INT_PIN_CONF, &m_bma400_dev);
+    bma400_check_rslt("bma400_set_device_conf", rslt);
+}
+//                       
+int8_t BMA400LL::i2c_write(uint8_t reg_addr, const uint8_t *reg_data, uint32_t length, void *intf_ptr)
+{
+    nrfx_err_t err_code;
+    BMA400LL& device = BMA400LLManager::lookup_device(0); //todo  Define intf_ptr to include chip_id
+
+    if (!length)
+        return BMA400_OK;
+    
+    if (length > BMA400_MAX_BUFFER_LEN + sizeof(reg_addr))
+        return BMA400_E_OUT_OF_RANGE;
+    
+    uint8_t buffer[BMA400_MAX_BUFFER_LEN];
+    buffer[0] = reg_addr;
+    memcpy(&buffer[1], reg_data, length);
+    
+    nrfx_twim_xfer_desc_t xfer = NRFX_TWIM_XFER_DESC_TX(device.m_addr, (uint8_t*)buffer, length + sizeof(reg_addr));
+
+    err_code = nrfx_twim_xfer(&BSP::I2C_Inits[device.m_bus].twim, &xfer, 0);
+    //todo: what does this need?
+//    err_code = nrfx_twim_tx(&BSP::I2C_Inits[device.m_bus].twim, device.m_addr, buffer, length + sizeof(reg_addr), false);
+    if (err_code != NRFX_SUCCESS)
+        return BMA400_E_COM_FAIL;
+
+    return BMA400_OK;
+}
+
+int8_t BMA400LL::i2c_read(uint8_t reg_addr, uint8_t *reg_data, uint32_t length, void *intf_ptr)
+{
+    nrfx_err_t err_code;
+    BMA400LL& device = BMA400LLManager::lookup_device(0); //todo  Define intf_ptr to include chip_id
+
+    nrfx_twim_xfer_desc_t xfer = NRFX_TWIM_XFER_DESC_TXRX(device.m_addr, &reg_addr, sizeof(reg_addr), (uint8_t*)reg_data, length);
+
+    err_code = nrfx_twim_xfer(&BSP::I2C_Inits[device.m_bus].twim, &xfer, 0);
+
+    // err_code = nrfx_twim_tx(&BSP::I2C_Inits[device.m_bus].twim, device.m_addr, &reg_addr, sizeof(reg_addr), true);
+    // if (err_code != NRFX_SUCCESS)
+    //     return BMA400_E_COM_FAIL;
+    
+    // err_code = nrfx_twim_rx(&BSP::I2C_Inits[device.m_bus].twim, device.m_addr, reg_data, length);
+    if (err_code != NRFX_SUCCESS)
+        return BMA400_E_COM_FAIL;
+
+    return BMA400_OK;
```

Comment:
In this case I would use the `NrfI2C::write` (with `no_stop=true`) and `NrfI2C::read` following the example of the other drivers.